src/App.js

```diff
@@ -0,0 +1,346 @@
+import Maplet from './Map.js'
+import Sidebar from './Sidebar.js'
+import Footer from './Footer.js'
+import MyModal from './MyModal.js'
+import borderData from './border.js'
+import leafletPip from 'leaflet-pip'
+import L from 'leaflet'
+
+import './App.css';
+
+Modal.setAppElement(MyModal)
+
+// global variables for storing game's random lat and long
+
+let randomLat
+let randomLong
+
+// App component with the majority of game's functionality and data
+
+class App extends React.Component {
+
+    constructor(props) {
+        super(props)
+
+        this.state = {
+
+            gameStarted: false,
+
+            score: 100,
+
+            gjLayer: L.geoJSON(borderData),
+
+            // geographical and political info
+
+            currentPoint: {
+                lat: 1,
+                lng: 1
+            },
+
+            startingPoint: {
+                lat: 1,
+                lng: 1
+            },
+
+            pathArray: [],
+
+            countyName: '',
+
+            town: '',
+
+            // buttons 
+
+            startDisabled: false,
+            guessDisabled: true,
+            quitDisabled: true,
+            returnDisabled: true,
+            northDisabled: true,
+            southDisabled: true,
+            eastDisabled: true,
+            westDisabled: true,
+
+            // modal keys
+
+            modalDisplayed: false,
+
+            modalContent: '',
+
+            countyChosen: ''
+        }
+    }
+
+    // random number generator for lat and long coordinates
+
+    randomInt = (max, min) => {
+        return (min + (Math.random() * (max - min)))
+    }
+
+    // function to check if random point is within Vermont polygon    
+
+    checkPoint = () => {
+
+        randomLong = this.randomInt(-71.51022535353107, -73.42613118833583)
+
+        randomLat = this.randomInt(45.007561302382754, 42.730315121762715)
+
+        let pointInVermont = leafletPip.pointInLayer([randomLong, randomLat], this.state.gjLayer)
+
+        // loop to keep generating random points until one falls within Vermont 
+
+        while (pointInVermont.length < 1) {
+
+            randomLong = this.randomInt(-71.51022535353107, -73.42613118833583)
+            randomLat = this.randomInt(45.007561302382754, 42.730315121762715)
+            pointInVermont = leafletPip.pointInLayer([randomLong, randomLat], this.state.gjLayer)
+        }
+
+        // once an appropriate point is found, coordinates passed into state ...
+
+        this.setState({
+            currentPoint: {
+                lat: randomLat,
+                lng: randomLong
+            },
+            startingPoint: {
+                lat: randomLat,
+                lng: randomLong
+            },
+        });
+
+        // ... the array of moves is given a starting point ...
+
+        this.setState(state => {
+            let pathArray = state.pathArray.concat(state.startingPoint);
+
+            return {
+                pathArray
+            }
```

Comment:
Another way you can do this is like this:
```js
  this.setState(state => {
    return { pathArray: state.pathArray.concat(state.startingPoint) }
  }
```
This does the same thing, but is just a little bit more concise.